### ISSUE TITLE
Siege 3.0.8 dramatically more thruput compared to 4.0.2

### ISSUE THREAD

**Issue by ralfbergs** [JoeDog/siege#88]
I run automatic benchmarks for my VDSL line every five minutes and create RRD graphs from it. Recently I upgraded from Ubuntu 16.04 (Siege version 3.0.8) to Ubuntu 16.10 (Siege 4.0.2), and suddenly my benchmark showed dramatically lower thruput rates (from 94 Mbit/s down to 60-70 Mbit/s).

I found that this was due to the Siege version that changed with the upgrade as described above.

[This is how I do my benchmark](https://github.com/ralfbergs/NetBench/blob/master/netbench.pl). Only the measured download speed changed, uplink was (virtually?) unchanged.

Any idea what's going on?

**Comment by JoeDog** [comment]
My hunch is that it's related to parsing. In the past, siege just read
everything into a buffer then discarded it. Now it will uncode compressed
content and parse it. I might be capturing some of that in the download
time.  Could you try it again with the parser off? (You can turn it off in
your siegerc file)

Jeff

**Comment by ralfbergs** [comment]
@JoeDog, thanks for your response.

Unfortunately turning the parser off didn't help. Downlink speed still below 70 Mbit/s. I immediately rolled back to 3.0.8, and speed went up to 89 Mbit/s...

Any other idea?

**Comment by JoeDog** [comment]
The only other thing that changed was chunked encoding. (Siege would often
ignore it because it didn't care about the contents.) Could you set the
protocol to 1.0 and compare the two? (You can set that in your siegerc.
Search for protocol

**Comment by ralfbergs** [comment]
@JoeDog, thanks for the new suggestion.

Unfortunately this didn't help either. Speed is still degraded. :-(

_Yet_ another idea? :-)

**Comment by JoeDog** [comment]
No. I have a feeling that some of the new features that can't be turned off
explain the difference.  You might want to try comparing just an image url.

**Comment by JoeDog** [comment]
Is it possible you're running the new version with a delay? 

Try running both with -b, i.e.,   siege -b -c10 -r10 URL

**Comment by ralfbergs** [comment]
@JoeDog I have always been using this "benchmark" switch... :-)

You can see it in my first post, there's a link to my Perl script that "drives" Siege.
